### DISCUSSION TITLE
Multipart body length limit 134217728 exceeded

### DISCUSSION THREAD

**Discussion by enfJoao** [oqtane/oqtane.framework#4707] [Q&A]
I'm trying to upload a 200Mb app into 2sxc on Oqtane.
Got all the configuration size limit errors fixable in IIS, but now I'm at 
"Message":"Multipart body length limit 134217728 exceeded.","Data":{},"InnerException":null,"HelpLink":null,"Source":"Microsoft.AspNetCore.WebUtilities"
Which seem oqtane related.
Is this fixable on my end?

**Comment by thabaum** [comment]
@enfJoao hello, while I have this fired up today I will take a look at this.  I see you have a cookie issue as well with 2sxc so I will install this module and be sure nothing relating to a previous PR I had submitted/merged earlier has anything to do with this.  It maybe hard to reproduce without a 200MB app of my own to upload using 2sxc.  @iJungleboy may know a little more about this?

**Comment by enfJoao** [comment]
You can place any 200MB file in any app and export it. The exported app will include all files in the folder.

The same way I removed the large files from my app and placed them again after import to bypass this issue.

The cookie issue may be a consequence of comipling errors and not importing libraries properly.

**Reply by thabaum** [reply]
@enfJoao can you test adding this to `Startup.cs` under ConfigureServices?
```
        public void ConfigureServices(IServiceCollection services)
        {
            ...
            services.Configure<FormOptions>(options =>
            {
                options.MultipartBodyLengthLimit = long.MaxValue;
            });

            services.Configure<KestrelServerOptions>(options =>
            {
                options.Limits.MaxRequestBodySize = long.MaxValue;
            });
            ...
         }
```

and for IIS something like this in your web.config

```
<configuration>
  <system.webServer>
    <security>
      <requestFiltering>
        <requestLimits maxAllowedContentLength="2147483648" /> <!-- 2 GB -->
      </requestFiltering>
    </security>
  </system.webServer>
</configuration>
```

Just an idea maybe you can test today?

**Comment by enfJoao** [comment]
Sure. Tomorrow I'll give a try.

**Comment by enfJoao** [comment]
Can you give me an exact location for that .cs? Is it a package or repo file? I don't use vs, so I only install from the release packages.

**Reply by thabaum** [reply]
@enfJoao This section would be located in your Startup.cs in the root directory of the Oqtane.Server project.

https://github.com/oqtane/oqtane.framework/blob/dev/Oqtane.Server/Startup.cs

I am not sure if you have done this before you can follow directions in the readme to fire this up using Visual Studio.

You would add this inside the Startup.cs file.  After most stuff loads up or anywhere it should set this setting as long as it is inside the         `public void ConfigureServices(IServiceCollection services) { }` method.

This would set Oqtane to use these configuration settings for these services on startup.

You can build the project and then install Oqtane with a basic setup on your developer machine and start editing code testing things.  Build the project first then run it as shown.  Oqtane Documentation also helps you understand how to perform this.  

Since Oqtane is a developer first framework we as a community work hard to keep it a great starting template for development.

https://docs.oqtane.org/manuals/admin/installation/development.html 

Simply add these changes, install Oqtane using localDB after running it from Visual Studio.  Install the module you like to work with to test a solution (in this case 2sxc module) as you would in production release.

When you build in release mode it puts all the files you need to actually copy to production if you chose to and run it.  All the upgrade does is take these files and and overwrite them in your production site.  All but the appsettings.json file pretty much I believe as this keeps your configuration such as your database connection string to allow Oqtane to run on the instance you already created.   Docs will let you know where these folders are as they can be in the folder set to publish.  

Publishing directly to a site is not supported, you have to manually copy these files to a working production environment if you want to "branch" off your own direction for example if "breaking" changes for the community would hold you back.

 You simply can change this line to any database that has Oqtane installation and you run that instance of it.  If the database is blank or not set Oqtane will go through the installation process to help set that database up.
 
 This is what I would do to test your situation.  Sometimes it is faster to test yourself as it takes time for a community member to donate support.  We all appreciate any reports of issues as this is 100% community driven and since we all use this product we all have skin in the game and reasons to help support and improve it.  From developer module creators to end users.

Thanks for taking time to bring this to our attention.  I hope this message is clear to understand as I do my best to help others when I can.
